src/fasta.rs

```diff
@@ -1,209 +1,932 @@
+                    contents: "CCCC".to_string(),
+                    line_range: (3, 5),
+                },
+            ],
         );
     }
 
     #[test]
     fn test_fasta_mutiple_contents_multiline() {
-        let r = SimpleFastaParser::parse(">Virus1\nAAAA\nAAAA\n>Virus2\nCCCC\nCCCC\n".as_bytes())
-            .unwrap();
-        assert_eq!(
-            r,
+        assert_parse_with_all_settings(
+            ">Virus1\nAAAA\nAAAA\n>Virus2\nCCCC\nCCCC\n",
             vec![
-                ("Virus1".to_string(), "AAAAAAAA".to_string()),
-                ("Virus2".to_string(), "CCCCCCCC".to_string())
-            ]
+                FastaRecord {
+                    header: "Virus1".to_string(),
+                    contents: "AAAAAAAA".to_string(),
+                    line_range: (1, 4),
+                },
+                FastaRecord {
+                    header: "Virus2".to_string(),
+                    contents: "CCCCCCCC".to_string(),
+                    line_range: (4, 7),
+                },
+            ],
         );
     }
 
     #[test]
-    fn test_dna_fasta() {
-        let r: Vec<(String, DnaSequence)> =
-            DnaFastaParser::parse(">Virus1\nAAAA".as_bytes()).unwrap();
+    fn test_fasta_triple() {
+        assert_parse_with_all_settings(
+            ">Virus1\nAAAA\nAAAA\n>Virus2\nCCCC\nCCCC\n>Virus3\nCCCC\nCCCC\n",
+            vec![
+                FastaRecord {
+                    header: "Virus1".to_string(),
+                    contents: "AAAAAAAA".to_string(),
+                    line_range: (1, 4),
+                },
+                FastaRecord {
+                    header: "Virus2".to_string(),
+                    contents: "CCCCCCCC".to_string(),
+                    line_range: (4, 7),
+                },
+                FastaRecord {
+                    header: "Virus3".to_string(),
+                    contents: "CCCCCCCC".to_string(),
+                    line_range: (7, 10),
+                },
+            ],
+        );
+    }
+
+    #[test]
+    fn test_fasta_trailing_header_only() {
+        assert_parse_with_all_settings(
+            ">Virus1\nAAAA\nAAAA\n>Virus2\nCCCC\nCCCC\n>Virus3",
+            vec![
+                FastaRecord {
+                    header: "Virus1".to_string(),
+                    contents: "AAAAAAAA".to_string(),
+                    line_range: (1, 4),
+                },
+                FastaRecord {
+                    header: "Virus2".to_string(),
+                    contents: "CCCCCCCC".to_string(),
+                    line_range: (4, 7),
+                },
+                FastaRecord {
+                    header: "Virus3".to_string(),
+                    contents: "".to_string(),
+                    line_range: (7, 8),
+                },
+            ],
+        );
+    }
+
+    #[test]
+    fn test_fasta_header_concat() {
+        assert_parse_with_concatenate_headers(
+            ">a\n>b\ntest",
```

Comment:
this is incorrect. ">a\n>b\ntest" should become header: `>a` and content `btest`. This is no different from this:
```
> header
AAAA
BBBB
```